File: Javascript/3775.reverse-words-with-same-vowel-count [1].js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    function findVowelCount(s){
        const vowels = ['a', 'e', 'i', 'o', 'u']

        return s.split('').reduce((acc, ch) => acc + vowels.includes(ch), 0)
    }

    const words = s.split(' ')
    const vowelCount1 = findVowelCount(words[0])
    let ans = words[0]

    for(let i=1; i<words.length; i++){
        ans += ' '
        
        if(findVowelCount(words[i]) === vowelCount1){
            ans += words[i].split('').reverse().join('')
        }else{
            ans += words[i]
        }
    }

    return ans
};

File: CPP [3501-4000]/3775.reverse-words-with-same-vowel-count [1].cpp
class Solution {
private:
    int findVowelCount(string& s){
        int count = 0;
        for(char ch: s)
            if(ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u')
                count++;
        
        return count;
    }

public:
    // T.C.=O(n), S.C.=O(n)
    string reverseWords(string s) {
        istringstream iss(s);
        string word = "";
        int vowelCount1 = -1;
        string ans = "";

        while(iss >> word){
            if(vowelCount1 == -1)
                vowelCount1 = findVowelCount(word);
            else if(findVowelCount(word) == vowelCount1)
                reverse(word.begin(), word.end());
            
            ans += word + " ";
        }

        ans.pop_back();
        return ans;
    }
};